pyconcepticon/commands.py

```diff
@@ -539,6 +540,100 @@ def _get_missing(o, _):
             _get_mergers(clist, api)
 
 
+@command()
+def check_new(args):
+    """
+    Expects a well-formed concept list as input (i.e. tsv, 'ID',
+    'CONCEPTICON_ID', 'NUMBER', 'CONCEPTICON_GLOSS' columns, etc.) and tests
+    for a number of potential issues:
+        - mismatch between glosses and Concepticon IDs
+        - availability of glosses in Concepticon
+        - if proposed glosses (starting with !) don't have IDs (they shouldn't!)
+        - if glosses are mapped more than once
+        - if 'NUMBER' and 'ID' are unique for the respective concept list.
+
+    concepticon checknew path_to_conceptlist.tsv
+    """
+    list_to_check = read_dicts(args.args[0])
+    api = Concepticon(args.data)
+    con_glosses = {c.id: c.gloss for c in api.conceptsets.values()}
+
+    def _get_duplicates(to_check):
+        known_items = set()
+        known_items_add = known_items.add
+        return [(i, key) for i, key in enumerate(to_check) if
+                key in known_items or known_items_add(key)]
```

Comment:
Just a nit: In case this is not a tight loop run many times, this is probably premature optimization.
I would definitely place the `if` on the second line though.

Reply:
Cool, thanks!